### DISCUSSION TITLE
Display table & column comments in SQL Lab UI

### DISCUSSION THREAD

**Discussion by jdangerx** [apache/superset#29369] [Ideas]
Hi!

When I add comments to my SQL tables and columns, I would love to see a little tooltip in the SQL Lab sidebar that shows those comments.

It looks like maybe we used to do this - here's an [ancient PR](https://github.com/apache/superset/pull/10844).

However, now there are no tooltips!

<img width="404" alt="image" src="https://github.com/apache/superset/assets/2495794/561d59b1-5516-45e0-80ef-778081928563">

The `table_metadata` API still *does* return the column & table comments:

<img width="1277" alt="image" src="https://github.com/apache/superset/assets/2495794/76b98c48-7869-42d2-a3f7-d49194839f16">

However they don't appear to be actually used in the [`TableMetadata`](https://github.com/apache/superset/blob/b5a72e21f73808bcca94341f5a904753a589e67b/superset-frontend/src/hooks/apiResources/tables.ts#L75) or [`Column`](https://github.com/apache/superset/blob/b5a72e21f73808bcca94341f5a904753a589e67b/superset-frontend/src/hooks/apiResources/tables.ts#L69) types.

My questions are:
* is this a regression/bug that should be fixed? If so, I'm happy to give it a shot - let me know about any conventions/components/dev environment/testing stuff that I should look at before I start. I would probably post some sort of rough plan of attack before getting too deep into the weeds to make sure I'm going in the right direction.
* is this a feature request? If so, does that require further discussion/decision making around whether or not to do this?
* it seems like the `external_metadata_by_name` endpoint that the dataset "sync columns from source" button hits also returns column comments, which don't get displayed anywhere... should that be a separate feature request?

**Comment by sfirke** [comment]
Pinging @justinpark as someone who contributes a lot to SQL Lab - what do you think Justin?  I appreciate this discussion and your willingness to take on the work 🙏 If we don't hear otherwise, @jdangerx, let's plan on giving this the green light for you to tackle.

**Reply by jdangerx** [reply]
It's been about a week - I'm going to assume I can move forward on this. Here's my non-detailed plan, any pitfalls you foresee?

* set up dev environment using the [docker-compose approach](https://superset.apache.org/docs/contributing/development)
* use [Testing Guidelines](https://github.com/apache/superset/wiki/Testing-Guidelines-and-Best-Practices) to fumble through the testing set up, and test for tooltips to show up on hover
* make those tests pass, trying not to break any [Frontend Guidlines](https://github.com/apache/superset/wiki/Frontend-Style-Guidelines). Though I notice this page is 2 years old, is it out-of-date?

**Reply by sfirke** [reply]
Sounds good to me!  I'm not sure about the front end guidelines page.  I see @rusackas was the last one to edit it, he might have thoughts. But the Contributing page you found is, I believe, as up to date as anything.

**Reply by jdangerx** [reply]
Great, thanks! Due to various work prioritization things I probably won't get to it for another week at least - but it's on my list!

**Reply by djs2000** [reply]
Hi everyone! I noticed this interesting discussion about displaying table and column comments in SQL Lab UI, and I'm curious whether there has been any progress on this feature since August 2024. @jdangerx - were you able to start working on the implementation? I'd be interested in learning about any updates or if there are any ways the community can help move this forward. Thanks!

**Reply by jdangerx** [reply]
Hi @djs2000 ! I never got around to working on this as we decided to not make our users use SQL Lab after talking to some of them. It does still seem like a useful thing to do though!

**Comment by VladislavLytvyniuk** [comment]
It would be good to add comments in SQL Lab when DDL contains table and column descriptions

CREATE TABLE public.lv_example_table (
    id INT PRIMARY KEY,
    name TEXT,
    description TEXT,
    created_at TIMESTAMP
);

COMMENT ON TABLE public.lv_example_table IS 'Example table with description';
COMMENT ON COLUMN public.lv_example_table.id IS 'Unique record identifier';
COMMENT ON COLUMN public.lv_example_table.name IS 'Element name';
COMMENT ON COLUMN public.lv_example_table.description IS 'Detailed element description';
COMMENT ON COLUMN public.lv_example_table.created_at IS 'Record creation date and time';
